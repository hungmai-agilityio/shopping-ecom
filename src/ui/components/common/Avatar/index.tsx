import { clsx } from 'clsx';
import Image, { ImageProps } from 'next/image';

// Constants
import { avatarSizeClasses, SIZE } from '@/constants';

interface AvatarProps extends ImageProps {
  size?: SIZE.SMALL | SIZE.MEDIUM | SIZE.LARGE;
  styles?: string;
  circle?: boolean;
  onClick?: () => void;
}

const Avatar = ({
  src,
  alt,
  size = SIZE.SMALL,
  styles,
  circle,
  onClick
}: AvatarProps) => {
  const defaultAvatar = '/user.svg';

  return (
    <div
      className={clsx('relative', avatarSizeClasses[size], styles, {
        'cursor-pointer': onClick
      })}
    >
      <Image
        data-testid="avatar"
        onClick={onClick}
        src={src || defaultAvatar}
        alt={alt}
        fill
        className={clsx(
          'object-cover font-bold',
          circle && 'rounded-full',
          styles
        )}
      />
    </div>
  );
};

export default Avatar;
