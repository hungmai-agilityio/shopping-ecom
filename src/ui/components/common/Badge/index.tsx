interface BadgeProps {
  quantity: number;
  children: React.ReactNode;
}

const Badge = ({ quantity, children }: BadgeProps) => {
  return (
    <div className="relative">
      {quantity > 0 && (
        <span className="absolute -top-3 -right-3 h-6 w-6 rounded-full flex justify-center items-center bg-primary text-white text-[10px] cursor-pointer z-10 overflow-hidden font-semibold">
          {quantity > 99 ? '99+' : quantity}
        </span>
      )}
      {children}
    </div>
  );
};

export default Badge;
