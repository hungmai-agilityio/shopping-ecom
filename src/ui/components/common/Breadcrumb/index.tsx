'use client';

import React from 'react';
import { usePathname } from 'next/navigation';
import { clsx } from 'clsx';
import Link from 'next/link';

interface BreadcrumbProps {
  separator?: string;
  color?: string;
}

const Breadcrumb = ({
  separator = '/',
  color = 'text-gray-700'
}: BreadcrumbProps) => {
  const pathname = usePathname();
  const paths = ['Home', ...pathname.split('/').filter(Boolean)];
  const items = paths.length;

  const displayPaths =
    items > 3 ? [paths[0], '...', paths[items - 2], paths[items - 1]] : paths;

  return (
    <nav className={clsx('flex text-sm', color)}>
      {displayPaths.map((path, index) => (
        <div key={index} className="flex items-center">
          <Link
            href={`/${paths.slice(1, index + 1).join('/')}`}
            className={clsx(
              index === displayPaths.length - 1
                ? 'font-semibold text-black'
                : 'hover:underline'
            )}
          >
            {path}
          </Link>
          {index < displayPaths.length - 1 && (
            <span className="mx-2 text-gray-400">{separator}</span>
          )}
        </div>
      ))}
    </nav>
  );
};

export default Breadcrumb;
