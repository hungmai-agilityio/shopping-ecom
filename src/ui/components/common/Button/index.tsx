import { ButtonHTMLAttributes, ReactNode } from 'react';
import clsx from 'clsx';

// Constants
import {
  btnSizeClasses,
  btnTypeClasses,
  popping,
  SIZE,
  TYPE
} from '@/constants';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: TYPE.PRIMARY | TYPE.SECOND | TYPE.THIRD;
  children?: ReactNode;
  size?: SIZE.SMALL | SIZE.MEDIUM | SIZE.LARGE;
  styles?: string;
}

const Button = ({
  children,
  variant = TYPE.PRIMARY,
  size = SIZE.MEDIUM,
  disabled,
  onClick,
  styles,
  ...props
}: ButtonProps) => {
  return (
    <button
      className={clsx(
        `rounded font-medium md:text-base text-xs ${popping.className}`,
        btnSizeClasses[size],
        btnTypeClasses[variant],
        styles,
        disabled && 'cursor-not-allowed opacity-50'
      )}
      disabled={disabled}
      onClick={onClick}
      {...props}
    >
      {children}
    </button>
  );
};

export default Button;
