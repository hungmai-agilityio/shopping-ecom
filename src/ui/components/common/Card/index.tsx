import React from 'react';
import clsx from 'clsx';
import { popping } from '@/constants';

interface CardProps {
  children: React.ReactNode;
  styles?: string;
  title?: string;
}

const Card = ({ children, styles, title }: CardProps) => {
  return (
    <div
      className={clsx('rounded-sm overflow-hidden', popping.className, styles)}
    >
      {title && (
        <div className="border-t-2 border-r-2 border-2 p-4 font-medium">
          {title}
        </div>
      )}
      {children}
    </div>
  );
};

export default Card;
