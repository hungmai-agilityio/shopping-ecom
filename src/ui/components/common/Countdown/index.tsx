'use client';
import React, { useState, useEffect } from 'react';

// Components
import CountdownItem from '@/ui/components/common/Countdown/Item';

// Libs
import { convertToSeconds, formatNumber, formatTime } from '@/libs';
import { clsx } from 'clsx';

interface CountdownProps {
  days: number;
  hours: number;
  minutes: number;
  seconds: number;
  isCircle?: boolean;
}

const Countdown = ({
  days,
  hours,
  minutes,
  seconds,
  isCircle
}: CountdownProps) => {
  const [timer, setTimer] = useState<number>(
    convertToSeconds(days, hours, minutes, seconds)
  );

  useEffect(() => {
    const timer = setInterval(() => {
      setTimer((prev) => {
        if (prev <= 0) {
          return convertToSeconds(days, hours, minutes, seconds);
        }
        return prev - 1;
      });
    }, 1000);

    return () => clearInterval(timer);
  }, [days, hours, minutes, seconds]);

  const {
    days: day,
    hours: hour,
    minutes: minute,
    seconds: sec
  } = formatTime(timer);

  const colon = !isCircle && (
    <span className="text-primary lg:text-3xl mt-6">:</span>
  );
  return (
    <div className="flex space-x-4 text-center text-dark">
      <CountdownItem
        label="Days"
        value={formatNumber(day)}
        isCircle={isCircle}
      />

      {colon}
      <CountdownItem
        label="Hours"
        value={formatNumber(hour)}
        isCircle={isCircle}
      />
      {colon}
      <CountdownItem
        label="Minutes"
        value={formatNumber(minute)}
        isCircle={isCircle}
      />
      {colon}
      <CountdownItem
        label="Seconds"
        value={formatNumber(sec)}
        isCircle={isCircle}
      />
    </div>
  );
};

export default Countdown;
