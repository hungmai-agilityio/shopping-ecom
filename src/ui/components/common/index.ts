export { default as Avatar } from '@/ui/components/common/Avatar';
export { default as Button } from '@/ui/components/common/Button';
export { default as Icon } from '@/ui/components/common/Icon';
export { default as Social } from '@/ui/components/common/Icon/Social';
export { default as Card } from '@/ui/components/common/Card';
export { default as CardPromo } from '@/ui/components/common/Card/Promo';
export { default as CardSolution } from '@/ui/components/common/Card/Solution';
export { default as CardProduct } from '@/ui/components/common/Card/Product';
export { default as Tag } from '@/ui/components/common/Tag';
export { default as Badge } from '@/ui/components/common/Badge';
export { default as Breadcrumb } from '@/ui/components/common/Breadcrumb';
export { default as Navbar } from '@/ui/components/common/Navbar';
export { default as Input } from '@/ui/components/common/Input';
export { default as SendEmail } from '@/ui/components/common/Input/SendEmail';
export { default as Logo } from '@/ui/components/common/Logo';
export { default as List } from '@/ui/components/common/List';
export { default as Countdown } from '@/ui/components/common/Countdown';
export { default as Rating } from '@/ui/components/common/Rating';
export { default as SizePicker } from '@/ui/components/common/SizePicker';
export { default as ColorPicker } from '@/ui/components/common/ColorPicker';
export { default as Carousel } from '@/ui/components/common/Carousel';
export { default as Quantity } from '@/ui/components/common/Quantity';
export { default as Dropdown } from '@/ui/components/common/Dropdown';
export { default as Table } from '@/ui/components/common/Table';

export * from '@/ui/components/common/Table/Body';
export * from '@/ui/components/common/Table/Cell';
export * from '@/ui/components/common/Table/Head';
export * from '@/ui/components/common/Table/Row';
export * from '@/ui/components/common/Typo';
