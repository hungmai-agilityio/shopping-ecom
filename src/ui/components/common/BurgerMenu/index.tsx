'use client';

import { useState } from 'react';

// Components
import { Icon, Navbar } from '@/ui/components';

interface BurgerProps {
  items: Array<{ name: string; url?: string }>;
}

const BurgerMenu = ({ items }: BurgerProps) => {
  const [menuOpen, setMenuOpen] = useState<boolean>(false);

  // Toggle menu open/close state
  const toggleMenu = () => setMenuOpen(!menuOpen);

  // Close menu function
  const closeMenu = () => setMenuOpen(false);

  return (
    <>
      <Icon
        src="/menu.svg"
        alt="menu"
        width={32}
        height={32}
        priority
        onClick={toggleMenu}
      />
      <div
        className={`fixed top-0 left-0 w-full h-full bg-white z-50 p-6
          transition-transform duration-300 ease-in-out ${menuOpen ? 'translate-x-0' : '-translate-x-full'}`}
      >
        <div className="flex justify-end mb-8">
          <Icon
            src="/close.svg"
            alt="close"
            width={24}
            height={24}
            priority
            onClick={toggleMenu}
          />
        </div>
        <Navbar
          items={items}
          styles="flex flex-col gap-6 text-2xl text-center"
          onClick={closeMenu}
        />
      </div>
    </>
  );
};

export default BurgerMenu;
