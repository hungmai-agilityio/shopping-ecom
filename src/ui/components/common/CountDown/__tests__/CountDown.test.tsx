import { Countdown } from '@/ui/components/common';
import { render, screen } from '@testing-library/react';

describe('Countdown component', () => {
  test('Should render Countdown component', () => {
    render(<Countdown days={1} hours={2} minutes={12} seconds={11} />);

    const days = screen.getByText('01');
    const hours = screen.getByText('02');
    const minutes = screen.getByText('12');
    const seconds = screen.getByText('11');

    expect(days).toBeInTheDocument();
    expect(hours).toBeInTheDocument();
    expect(minutes).toBeInTheDocument();
    expect(seconds).toBeInTheDocument();
  });

  test('should be render match to snapshot', () => {
    const { container } = render(
      <Countdown days={1} hours={2} minutes={12} seconds={11} />
    );

    expect(container).toMatchSnapshot();
  });
});
