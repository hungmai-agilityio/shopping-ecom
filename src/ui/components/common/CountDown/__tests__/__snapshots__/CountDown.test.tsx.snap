// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown component should be render match to snapshot 1`] = `
<div>
  <div
    class="flex space-x-4 text-center text-dark"
  >
    <div
      class="flex flex-col items-center"
    >
      <span
        class="lg:text-lg text-xs font-medium"
      >
        Days
      </span>
      <span
        class="lg:text-3xl text-xl font-bold"
      >
        01
      </span>
    </div>
    <span
      class="text-primary lg:text-3xl mt-6"
    >
      :
    </span>
    <div
      class="flex flex-col items-center"
    >
      <span
        class="lg:text-lg text-xs font-medium"
      >
        Hours
      </span>
      <span
        class="lg:text-3xl text-xl font-bold"
      >
        02
      </span>
    </div>
    <span
      class="text-primary lg:text-3xl mt-6"
    >
      :
    </span>
    <div
      class="flex flex-col items-center"
    >
      <span
        class="lg:text-lg text-xs font-medium"
      >
        Minutes
      </span>
      <span
        class="lg:text-3xl text-xl font-bold"
      >
        12
      </span>
    </div>
    <span
      class="text-primary lg:text-3xl mt-6"
    >
      :
    </span>
    <div
      class="flex flex-col items-center"
    >
      <span
        class="lg:text-lg text-xs font-medium"
      >
        Seconds
      </span>
      <span
        class="lg:text-3xl text-xl font-bold"
      >
        11
      </span>
    </div>
  </div>
</div>
`;
